### PR TITLE
Refactor IoTeX staking.

### LINKED ISSUE #861: Refactor TWIoTeXStaking
Move the logic from C implementation to cpp + protobuf

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Refactor IoTeX staking, Move the logic from C implementation to cpp.  Fixes #861 .

## Testing instructions

Unit tests *oTe*.

## Types of changes

* Minor Refactor

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.

### CODE REVIEW

File: include/TrustWalletCore/TWIoTeXStaking.h [hunk 1]
```diff
@@ -15,23 +15,23 @@ TW_EXTERN_C_BEGIN
 TW_EXPORT_CLASS
 struct TWIoTeXStaking;
 
-/// Function to generate Stake message
+/// Function to generate Stake message.  Result must be freed.
 TW_EXPORT_STATIC_METHOD
```
Comment (vikmeup) [inline include/TrustWalletCore/TWIoTeXStaking.h]: is this freeding must happen for iOS/Android interfaces or internal C++ code?

Comment (hewigovens) [reply include/TrustWalletCore/TWIoTeXStaking.h]: let's remove this C interface instead? add a few staking message type to proto

Comment (hewigovens) [reply include/TrustWalletCore/TWIoTeXStaking.h]: like Cosmos

Comment (catenocrypt) [reply include/TrustWalletCore/TWIoTeXStaking.h]: > is this freeding must happen for iOS/Android interfaces or internal C++ code?

In the swift&jni wrappers it is freed automatically, I think.

Comment (hewigovens) [reply include/TrustWalletCore/TWIoTeXStaking.h]: if not necessary, let's remove this C interface? I can do that later after binary xcframework if you like

**Pull request merged**